beacon-chain/core/altair/deposit.go

```diff
@@ -5,11 +5,32 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/helpers"
 	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
 	"github.com/prysmaticlabs/prysm/v5/encoding/bytesutil"
 	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
 )
 
+// ProcessPreGenesisDeposits processes a deposit for the beacon state before chainstart.
```

Comment:
Why are these in `altair`? I think these additions are not part of altair hard fork right

Reply:
the alternative would be to break out process deposits into yet another for phase 0. I considered this but just decided to have the two ( altair and electra). if you think I should still break this out I can. 
the issue with leaving it in the blocks package is the circular dependency, processPreGenesisDeposits calls process deposits which needs to specify a fork for process deposits, in this case it only makes sense to use the current version of process deposits